packages/htmlbars-syntax/lib/utils.js

```diff
@@ -17,7 +17,14 @@ export function parseComponentBlockParams(element, program) {
     attrNames.push(element.attributes[i].name);
   }
 
-  var asIndex = attrNames.indexOf('as');
+  var asIndex = -1;
+
+  for (i = 0; i < attrNames.length; i++) {
+    if (attrNames[i] === 'as') {
+      asIndex = i;
+      break;
+    }
+  }
```

Comment:
@salzhrani I encourage you to create an `indexOfArray` method that is the native version if available, and a poly-fill if not.

I suggest reviewing [ember's implementation](https://github.com/emberjs/ember.js/blob/d677cbb0c51eee34910ab4075bc53c938d0e3055/packages/ember-metal/lib/array.js#L62) for inspiration!